File: java/DVTuple.java
package JThrustRTC;

public class DVTuple extends DeviceViewable
{
	private static long create(DeviceViewable[] objs, String[] name_objs)
	{		
		long[] p_objs = new long[objs.length];
		for (int i = 0; i<objs.length; i++ )
			p_objs[i] = objs[i].cptr();
		return Native.dvtuple_create(p_objs, name_objs);
	}

	public DVTuple(DeviceViewable[] objs, String[] name_objs)
	{
		super(create(objs, name_objs));
		m_objs = objs;
	}

    private DeviceViewable[] m_objs;

    @Override
    public void close()
    {
    	m_objs = null;
        super.close();
    }
}

File: CSharp/ThrustRTCSharp/DVTuple.cs
using System;
using ThrustRTCLR;
using System.Runtime.InteropServices;

namespace ThrustRTCSharp
{
    public class DVTuple : DeviceViewable
    {
        private static IntPtr create(DeviceViewable[] objs, string[] name_objs)
        {
            CapturedDeviceViewable_clr[] arg_map = new CapturedDeviceViewable_clr[objs.Length];
            for (int i = 0; i < objs.Length; i++)
            {
                arg_map[i].obj = objs[i].m_cptr;
                arg_map[i].obj_name = Marshal.StringToHGlobalAnsi(name_objs[i]);
            }

            return Native.dvtuple_create(arg_map);
        }

        public DVTuple(DeviceViewable[] objs, string[] name_objs) : base(create(objs, name_objs))
        {
            m_objs = objs;
        }

        protected override void Dispose(bool disposing)
        {
            if (disposed) return;
            m_objs = null;
            base.Dispose(disposing);
        }

        private DeviceViewable[] m_objs;
    }
}
